vaadin-combo-box-flow/src/main/resources/META-INF/resources/frontend/comboBoxConnector.js

```diff
@@ -25,12 +25,57 @@ window.Vaadin.Flow.comboBoxConnector = {
 
     comboBox.$connector = {};
 
+    // holds pageIndex -> callback pairs of subsequent indexes (current active range)
     let pageCallbacks = {};
     let cache = {};
     let lastFilter = '';
+    const placeHolder = new Vaadin.ComboBoxPlaceholder();
+    const MAX_RANGE_COUNT = Math.max(comboBox.pageSize * 2, 150); // Max item count in active range
+
+    // From https://codeburst.io/throttling-and-debouncing-in-javascript-b01cad5c8edf
+    const throttle = (func, limit) => {
+      let lastFunc
+      let lastRan
+      return function() {
+        const context = this
+        const args = arguments
+        if (!lastRan) {
+          func.apply(context, args)
+          lastRan = Date.now()
+        } else {
+          clearTimeout(lastFunc)
+          lastFunc = setTimeout(function() {
+            if ((Date.now() - lastRan) >= limit) {
+              func.apply(context, args)
+              lastRan = Date.now()
+            }
+          }, limit - (Date.now() - lastRan))
+        }
+      }
+    };
+    let setRequestedRange;
+    const rangeRequestThrottle = throttle(() => setRequestedRange && setRequestedRange(), 50);
```

Comment:
The optimal behavior would probably be debouncing after all, but so that if there's no active timeout, it would make the first call immediately and start the timeout.

Also, the timeout could be bigger. Requests are delayed only when scrolling really fast, and having e.g. 200ms would avoid a lot more requests, without adding too much (IMO) delay to the fast-scrolling case.